### ISSUE TITLE
model gets 100% validation accuracy in the first epoch after keyboard interrupting previous model training.

### ISSUE THREAD

**Issue by tomasj12** [tensorflow/tensorflow#56702]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Support

### Source

binary

### Tensorflow Version

2.9.1

### Custom Code

No

### OS Platform and Distribution

Ubuntu 20.04

### Python version

3.9

### GPU model and memory

AMD RADEON 

### Current Behaviour?

```shell
I train a model for 20 epochs, after that, I interrupt training with the stop button in jupyter notebook. After rerunning the model fitting, the model gets validation 100% accuracy in the first epoch.

I expected that the model would train from scratch after keyboard interruption during the previous training.
```

### Standalone code to reproduce the issue

```shell
# nothing special
# please, the code bellow is not important...

inputs = tf.keras.layers.Input(shape = intput_shape)
x = tf.keras.layers.Dense(128)(inputs)
outputs = tf.keras.layers.Dense(1, activation='sigmoid')(x)

model = tf.keras.Model(inputs=inputs,outputs=outputs)
model.compile(loss=tf.keras.losses.BinaryCrossEntropy(), optimizer='adam',metrics='acc')
tb_callbacks = tf.keras.callbacks.TensorBoard('./logs/')
model.fit(train_data,validation_set=val_data,epochs=100)
# fitting...
# here, after 30 epochs, val_accuracy is around 95%, I interrupt a training process with stop button in the jupyter notebook
# run again fitting
model.fit(train_data,validation_set=val_data,epochs=100)
# 100% val_accuracy after first epoch
```

### Relevant log output

_No response_</details>

**Comment by mohantym** [comment]
Hi @tomasj12 !
You can just use initial epoch in model fit like below to start training from next epoch.
```
inputs = tf.keras.layers.Input(shape = intput_shape)
x = tf.keras.layers.Dense(128)(inputs)
outputs = tf.keras.layers.Dense(1, activation='sigmoid')(x)

model = tf.keras.Model(inputs=inputs,outputs=outputs)
model.compile(loss=tf.keras.losses.BinaryCrossEntropy(), optimizer='adam',metrics='acc')
tb_callbacks = tf.keras.callbacks.TensorBoard('./logs/')
model.fit(train_data,validation_set=val_data,epochs=100)
# fitting...
# here, after 30 epochs, val_accuracy is around 95%, I interrupt a training process with stop button in the jupyter notebook
# run again fitting
model.fit(train_data,validation_set=val_data, initial_epoch=31,epochs=100)
```
Thank you!
